### PR TITLE
USWDS-Site: Add known issues section to memorable date page

### LINKED ISSUE #2682: USWDS - Site - Known issues: Add known issues to memorable date component
Usability test findings from the 'zebra' batch, which included testing memorable date, revealed some issues that we should add to the 'known issues' section for memorable date. We should also add an alert.

Issues:

- [USWDS - Memorable date: Input formatting requirements are unclear to users of screen readers #5902](https://github.com/uswds/uswds/issues/5902)
- [USWDS - Memorable date: Dropdown search functionality doesn't meet some users' expectations. #5910](https://github.com/uswds/uswds/issues/5910)
- [USWDS - Memorable date: Dropdown options containing month name and number can be confusing for screen reader users. #5911](https://github.com/uswds/uswds/issues/5911)

### PR DESCRIPTION
**Pull request by amyleadem**
# Summary

Added a known issues section to the memorable date component page.

> [!Important]
> We should update the changelog dates before merge

> [!Note]
> The build errors will resolve when we merge either #2721 or #2731.

## Related issue
Closes #2682 

## Preview link

[Memorable date page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-known-issues-mem-date/components/memorable-date/)

## Testing and review

- Confirm a known issues section appears on the memorable date page
- Confirm the info alert is present at the top of the memorable date page
- On the known issues list item:
  - Confirm that the issue descriptions are accurate
  - Confirm that GitHub links work as expected
  - Confirm there are no spelling or grammar issues
- Confirm the changelog entry is present and accurate

### CODE REVIEW

**Review by jaclinec — CHANGES_REQUESTED** [review]
Everything looks great - I just have one correction to suggest before merging.

The second issue copy should be changed from "Users expected to be able to search the month input by either name or number but could do neither." to "Users expected to be able to search the month input by either name or number **but could only search by number.**"

**Comment by amyleadem** [comment]
Thanks! I updated the summary in [704f417](https://github.com/uswds/uswds-site/pull/2716/commits/704f4172cf165c67eb0ac31ef5d3d6de5a81c5a5). Let me know if you need any other changes.

**Review by jaclinec — APPROVED** [review]
Perfect - approved! Thanks!

**Comment by amyleadem** [comment]
@finekatie and @mahoneycm This is ready for your review!

**Review by finekatie — COMMENTED** [review]
I'm not seeing a new row for the change log - does that show up after you publish?

Wondering about the phrase "users experienced friction" in the third issue. I would suggest something plainer, but I see the same language is in the GH issue, so I'll leave it be. 

Other than these two things, LGTM :)

**Review by mahoneycm — APPROVED** [review]
LGTM!

**Comment by amyleadem** [comment]
@finekatie It looked like we had a build failure on the page preview. The changelog should be visible now. Let me know if you need any changes!

**Comment by finekatie** [comment]
> @finekatie It looked like we had a build failure on the page preview. The changelog should be visible now. Let me know if you need any changes!

Sorry - I think I re-requested review by accident? I approve and LGTM :)

**Review by finekatie — APPROVED** [review]
LGTM!

File: _data/issues/memorable-date.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+- date: 2024-05-03
+  summary: Screen reader users were confused that the component announced both the month's number and name (e.g., "01 - January").
+  issue_repo: uswds
+  issue_number: 5911
+- date: 2024-05-03
+  summary: Users expected to be able to search the month input by either name or number but could only search by number.
+  summary_additional:
+  issue_repo: uswds
+  issue_number: 5910
+- date: 2024-04-29
```
[the remark this answers is not in the document]
Comment (thisisdano) [reply _data/issues/memorable-date.yml:13]: These issues are closed

File: _data/issues/memorable-date.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+title: Memorable date
+alert_type: info
+items:
+  summary: Screen reader users experienced friction with understanding what formatting was needed for the component.
```
Comment (thisisdano) [inline _data/issues/memorable-date.yml:4]:
```suggestion
  summary: Screen reader users experienced friction with understanding formatting requirements for each input.
```

**Review by thisisdano — APPROVED** [review]

**Pull request merged**